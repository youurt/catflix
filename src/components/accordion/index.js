import React, { useState, useContext, createContext } from 'react';
import {
  Container,
  Inner,
  Title,
  Item,
  Header,
  Body,
  Frame,
} from './styles/accordion';

const ToggleContext = createContext();

const Accordion = ({ children, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};
Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};

Accordion.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Item = ({ children, ...restProps }) => {
  const [toggleShow, setToggleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = ({ children, ...restProps }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header
      onClick={() => setToggleShow((prevToggleShow) => !prevToggleShow)}
      {...restProps}
    >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = ({ children, ...restProps }) => {
  const { toggleShow } = useContext(ToggleContext);

  return (
    <Body className={toggleShow ? 'open' : 'closed'} {...restProps}>
      <span>{children}</span>
    </Body>
  );
};

export default Accordion;
