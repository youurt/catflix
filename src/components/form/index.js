import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from './styles/form';

const Form = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Form.Base = ({ children, ...restProps }) => {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = ({ children, ...restProps }) => {
  return <Error {...restProps}>{children}</Error>;
};

Form.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Form.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

Form.TextSmall = ({ children, ...restProps }) => {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

Form.Link = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

Form.Input = ({ children, ...restProps }) => {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = ({ children, ...restProps }) => {
  return <Submit {...restProps}>{children}</Submit>;
};

export default Form;
