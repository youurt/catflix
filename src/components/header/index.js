import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import { Background, Container, Logo, ButtonLink } from './styles/header';

const Header = ({ bg = true, children, ...restProps }) => {
  return bg ? <Background {...restProps}>{children}</Background> : children;
};

Header.Frame = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = ({ children, ...restProps }) => {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = ({ to, ...restProps }) => {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

export default Header;
