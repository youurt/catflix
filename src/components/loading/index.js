import React from 'react';
import { Spinner, LockBody, Picture, ReleaseBody } from './styles/loading';

const Loading = ({ src, ...restProps }) => {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid="loading-picture" />
    </Spinner>
  );
};

Loading.ReleaseBody = () => {
  return <ReleaseBody />;
};

export default Loading;
