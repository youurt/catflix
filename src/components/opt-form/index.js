import React from 'react';
import { Container, Input, Button, Text, Break } from './styles/opt-form';

const OptForm = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

OptForm.Input = ({ ...restProps }) => {
  return <Input {...restProps} />;
};

OptForm.Button = ({ children, ...restProps }) => {
  return (
    <Button {...restProps}>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try Now!" />
    </Button>
  );
};

OptForm.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function OptFormBreak({ ...restProps }) {
  return <Break {...restProps} />;
};

export default OptForm;
