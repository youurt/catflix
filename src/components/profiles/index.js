import React from 'react';
import { Container, Title, List, Picture, Name, Item } from './styles/profiles';

const Profiles = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Profiles.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Profiles.List = ({ children, ...restProps }) => {
  return <List {...restProps}>{children}</List>;
};

Profiles.User = ({ children, ...restProps }) => {
  return <Item {...restProps}>{children}</Item>;
};

Profiles.Name = ({ children, ...restProps }) => {
  return <Name {...restProps}>{children}</Name>;
};

Profiles.Picture = ({ src, ...restProps }) => {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`}
    />
  );
};

export default Profiles;
