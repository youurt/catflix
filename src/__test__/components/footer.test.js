import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from '../../components';

describe('<Footer/>', () => {
  it('should render the <Footer/> with populated data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact Us!</Footer.Title>
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Ways to watch</Footer.Link>
            <Footer.Link href="#">Coporate Information</Footer.Link>
            <Footer.Link href="#">Classicflix Originals</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href="#">Help Center</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Redeem Gift Cards</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href="#">Media Center</Footer.Link>
            <Footer.Link href="#">Buy Gift Card</Footer.Link>
            <Footer.Link href="#">Cookie Preferences</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Ugur Tigu</Footer.Text>
      </Footer>
    );

    expect(getByText('Questions? Contact Us!')).toBeTruthy();
    expect(getByText('FAQ')).toBeTruthy();
    expect(getByText('Ways to watch')).toBeTruthy();
    expect(getByText('Coporate Information')).toBeTruthy();
    expect(getByText('Classicflix Originals')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
