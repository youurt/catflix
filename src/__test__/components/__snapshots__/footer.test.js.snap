// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render the <Footer/> with populated data 1`] = `
<div
  class="footer__Container-sc-14iizrq-0 dGvmnd"
>
  <div
    class="footer__Title-sc-14iizrq-4 TyDKQ"
  >
    Questions? Contact Us!
  </div>
  <div
    class="footer__Row-sc-14iizrq-2 hxdcGR"
  >
    <div
      class="footer__Column-sc-14iizrq-1 dgmiLF"
    >
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        FAQ
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Ways to watch
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Coporate Information
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Classicflix Originals
      </div>
    </div>
    <div
      class="footer__Column-sc-14iizrq-1 dgmiLF"
    >
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Help Center
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Jobs
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Terms of Use
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Contact Us
      </div>
    </div>
    <div
      class="footer__Column-sc-14iizrq-1 dgmiLF"
    >
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Account
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Redeem Gift Cards
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Privacy
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Speed Test
      </div>
    </div>
    <div
      class="footer__Column-sc-14iizrq-1 dgmiLF"
    >
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Media Center
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Buy Gift Card
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Cookie Preferences
      </div>
      <div
        class="footer__Link-sc-14iizrq-3 hAhwQV"
        href="#"
      >
        Legal Notices
      </div>
    </div>
  </div>
  <div
    class="footer__Break-sc-14iizrq-6 fhuiAY"
  />
  <div
    class="footer__Text-sc-14iizrq-5 dQDtJh"
  >
    Ugur Tigu
  </div>
</div>
`;
